plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt

```diff
@@ -99,10 +99,17 @@ class GesturesPluginImpl : GesturesPlugin, GesturesSettingsBase, MapStyleObserve
   private val scheduledAnimators = ArrayList<ValueAnimator>()
   private var gesturesInterpolator = LinearOutSlowInInterpolator()
 
-  // needed most likely for devices with API <= 23 only
-  // duration = 0 will still make animation end / cancel not immediately
-  // this may cause cancelling some easeTo animation without single camera update
+  // Devices with API <= 23 animating with duration = 0 will send initial value from Looper with
+  // some small delay as a result.
+  // Hence multiple immediate animations sent close to each other may cancel previous ones.
+  // This becomes visible with move gesture since animation steps are relative to each other.
+  // So we defer move distances that are not applied to apply later (when immediate ease will end).
+  // Does not apply to shove/scale/rotate gestures since they use value from detector that is
+  // relative to the animation start.
   private var immediateEaseInProcess = false
```

Comment:
`onAnimationEnd` callback is fired from the Handler of the main thread, thus busy thread or some system lag may keep the `immediateEaseInProcess` flag set until the next onMove gesture arrives and is skipped because of the flag.